### ISSUE TITLE
Question about the output dimension of Autoencoder

### ISSUE THREAD

**Issue by ZihanChen1995** [rusty1s/pytorch_geometric#1300]
## ❓ Questions & Help

Hi Matthias, thank you for this handy library. I'm trying to use the Autoencoder ([autoencoder.py](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/autoencoder.py)) for the node embeddings, but I found a question about the output dimension. Could you please help me with this?

I use the `CiteSeer` dataset with `split = full`. (`-->` represents output)
```
dataset = Planetoid("/tmp/citeseer", "CiteSeer", split = 'full', transform=T.NormalizeFeatures())
data = dataset[0]
data
--> Data(edge_index=[2, 9104], test_mask=[3327], train_mask=[3327], val_mask=[3327], x=[3327, 3703], y=[3327])


data.edge_index
-->tensor([[   0,    1,    1,  ..., 3324, 3325, 3326],
        [ 628,  158,  486,  ..., 2820, 1643,   33]])
```
Then I just follow the steps, use `train_test_split_edges` to split the data, turn labels and masks into `None`, use `train_pos_edge_index` to train the model, and finally encode the whole dataset.

The output of dataset looks like the following:
```
model.eval()
z = model.encode(x, original_data.edge_index.to(dev))
z.shape

--> torch.Size([3327, 16])
```

As you can see, we have these tensors:

- feature matrix (x): [3327, 3703]
- node label (y): [3327]
- final output node embedding (z): [3327, 16]
* * *
**So my questions are:**

1. Does the row index in x, y, and z all have the same meaning, which points to the same node? (I think the answer is yes, just for a double-check with you)

2. If 1 is yes, does this mean "row index" strictly points to the node index in the original `edge_index`? For example, the first row in x, y, z all point to the number '0' in `edge_index`, while the second row means '1' in `edge_index`, and so on.

3. If I want to use a new edge list to train my model, do I need to put all the unique nodes in the first tensor of `edge_index` (`edge_index[0]`)? So later when I encode them into `[0, max(unique_node)]`, all the new index can appear in the first tensor firstly.

4. If I have some new nodes come, and I want to encode them using the existing model. When I implement `model.encode(x, new_edge_list)`, should I use the original `x` or the updated `x` that contains new nodes? And how can I insert the features of the new nodes into our existing model?

I'm sorry for these long questions, Matthias. Thank you so much in advance for your time and help. Have a nice day!

**Comment by rusty1s** [comment]
Hi,

1. Yes.
2. Yes.
3. I'm not sure I understand that. `edge_index` holds the source and target node indices of each edge. `unique(edge_index[0])` gives you all the nodes that have at least one outgoing edge. If you want to add edges to it, you can simply concatenate them to `edge_index`. They do not need to be sorted.
4. If you have new nodes, `x` needs to also hold node features for those new nodes. The easiest way is to just concatenate them to new original `x` in the first dimension.

**Comment by ZihanChen1995** [comment]
Hi Matthias! Thank you a lot for your reply. It's really helpful, and I understand it now.

Thanks again!
